ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -5,14 +5,21 @@ import utils 1.0
 
 import StatusQ.Core 0.1
 import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1 as SQUtils
 import StatusQ.Popups.Dialog 0.1
 
+import shared.popups.send.controls 1.0
+
 StatusDialog {
     id: root
-    title: qsTr("Swap")
 
-    // This should be the only property which should be used when being launched from elsewhere
-    property SwapFormData formData: SwapFormData {}
+    /* This should be the only property which should be used to input
+    parameters to the modal when being launched from elsewhere */
+    required property SwapInputParamsForm swapInputParamsForm
+    required property SwapModalAdapter swapAdaptor
```

Comment:
Why do you require the `SwapInputParamsForm swapFormData` twice? Once standalone, and once part of the `swapAdaptor`? Once should be enough :)

Reply:
The thing is I need to set swapInputParamsForm when launching the app by calling 
`openPopup(swapModal, {swapInputParamsForm: parameters})`
this is why I cannot encapsulate it within the adaptor.
the reason why it is needed in the adaptor is to be able to set model used to show account balances based on selection made in the swapInputParamsForm. Not sure how I can avoid this.. so I suggest to keep it the way it is